### PR TITLE
urllib3 v2+ support

### LINKED ISSUE #86: pook broken in urllib3>=2
reproducer:

```
import pook
import urllib3

@pook.on
def test_my_api():
    mock = pook.post('https://example.org', reply=200)
    http = urllib3.PoolManager(headers={'User-Agent': 'foo'})
    resp = http.request('POST', 'https://example.org')
    assert resp.status == 200
    assert mock.calls == 1
```

in urllib3<2, test passes, in urllib3>=2:

```
=========================================================== FAILURES ============================================================
__________________________________________________________ test_my_api __________________________________________________________

    @pook.on
    def test_my_api():
        mock = pook.post('https://example.org', reply=200)
        http = urllib3.PoolManager(headers={'User-Agent': 'foo'})
>       resp = http.request('POST', 'https://example.org')

test_thing.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.11/site-packages/urllib3/_request_methods.py:118: in request
    return self.request_encode_body(
.venv/lib/python3.11/site-packages/urllib3/_request_methods.py:217: in request_encode_body
    return self.urlopen(method, url, **extra_kw)
.venv/lib/python3.11/site-packages/urllib3/poolmanager.py:443: in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
.venv/lib/python3.11/site-packages/pook/interceptors/urllib3.py:184: in handler
    return self._on_request(urlopen, path, conn, method, url,
.venv/lib/python3.11/site-packages/pook/interceptors/urllib3.py:122: in _on_request
    real_headers = dict(headers or {})
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[TypeError('sequence item 0: expected str instance, bool found') raised in repr()] HTTPHeaderDict object at 0x10b2cca50>
key = '__urllib3_bypass__'

    def __getitem__(self, key: str) -> str:
        val = self._container[key.lower()]
>       return ", ".join(val[1:])
E       TypeError: sequence item 0: expected str instance, bool found

.venv/lib/python3.11/site-packages/urllib3/_collections.py:259: TypeError
==================================================== short test summary info ====================================================
FAILED test_thing.py::test_my_api - TypeError: sequence item 0: expected str instance, bool found
======================================================= 1 failed in 0.12s =======================================================
```

### PR DESCRIPTION
**Pull request by wimglenn**
Don't put non-strings into http header dict.

Closes #86

### CODE REVIEW

File: tests/unit/interceptors/urllib3_test.py [hunk 1]
```diff
@@ -6,12 +6,12 @@
 
 @pook.on
 def assert_chunked_response(input_data, expected):
-    (pook.get('httpbin.org/foo')
+    (pook.get('https://httpbin.org/foo')
```
Comment (sarayourfriend) [inline tests/unit/interceptors/urllib3_test.py:9]: Just to clarify, are these changes necessary for the fix as well?

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
@wimglenn This looks good to me, can you rebase the PR? Then it should be ready to merge :pray:

File: tests/unit/interceptors/urllib3_test.py (hunk 1 above)
Comment (wimglenn) [reply tests/unit/interceptors/urllib3_test.py:9]: They're necessary for v2+ support, yes. There is `DeprecationWarning` from urllib3 without that (see my commit messages)

**Review by sarayourfriend — APPROVED** [review]
LGTM, thanks!

**Comment by wimglenn** [comment]
@sarayourfriend  Can this be released to PyPI? I'm still installing directly from feature branch in test requirements of some other projects ..

**Comment by sarayourfriend** [comment]
I don't have the ability to cut releases but I will email h2non now.

**Comment by wimglenn** [comment]
@sarayourfriend  It just needs the `@pook.on` decorator on the test. Somehow I forgot that originally, oops.

**Comment by sarayourfriend** [comment]
Oh god, how did I forget that too! Thank you so much :facepalm:

**Comment by wimglenn** [comment]
@sarayourfriend  Hello, any update on a PyPI release?

**Comment by sarayourfriend** [comment]
I haven't heard back from h2non. I'll email him again today.

**Comment by sarayourfriend** [comment]
@wimglenn, h2non gave me permissions for cutting releases myself. The release is up under v1.2.0. This is my first time releasing code on PyPI so please let me know if you notice any issues and I will respond ASAP to resolve them. It should be faster to cut releases in the future now that I'm able to do so myself.

**Comment by wimglenn** [comment]
Thanks - I've moved one app over in https://github.com/wimglenn/advent-of-code-data/pull/135 without a hitch. PyPI's 1.2 seems fine to me.
